pkg/network/usm/procnet/parser.go

```diff
@@ -0,0 +1,223 @@
+//    00000150:00000000 01:00000019 00000000
+//       |        |     |     |       |--> number of unrecovered RTO timeouts
+//       |        |     |     |----------> number of jiffies until timer expires
+//       |        |     |----------------> timer_active (see below)
+//       |        |----------------------> receive-queue
+//       |-------------------------------> transmit-queue
+//
+//    1000        0 54165785 4 cd1e6040 25 4 27 3 -1
+//     |          |    |     |    |     |  | |  | |--> slow start size threshold,
+//     |          |    |     |    |     |  | |  |      or -1 if the threshold
+//     |          |    |     |    |     |  | |  |      is >= 0xFFFF
+//     |          |    |     |    |     |  | |  |----> sending congestion window
+//     |          |    |     |    |     |  | |-------> (ack.quick<<1)|ack.pingpong
+//     |          |    |     |    |     |  |---------> Predicted tick of soft clock
+//     |          |    |     |    |     |              (delayed ACK control data)
+//     |          |    |     |    |     |------------> retransmit timeout
+//     |          |    |     |    |------------------> location of socket in memory
+//     |          |    |     |-----------------------> socket reference count
+//     |          |    |-----------------------------> inode
+//     |          |----------------------------------> unanswered 0-window probes
+//     |---------------------------------------------> uid
+//
+// Source: https://www.kernel.org/doc/Documentation/networking/proc_net_tcp.txt
+
+type scanner struct {
+	file   *os.File
+	reader *bufio.Reader
+
+	hexDecodingBuffer []byte
+}
+
+func newScanner(filepath string) (*scanner, error) {
+	f, err := os.Open(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	reader := bufio.NewReader(f)
+
+	// Skip header line
+	_, _ = reader.ReadBytes('\n')
+
+	return &scanner{
+		file:              f,
+		reader:            reader,
+		hexDecodingBuffer: make([]byte, 16),
+	}, nil
+
+}
+
+func (s *scanner) Next() (entry, bool) {
+	b, err := s.reader.ReadBytes('\n')
+	if err != nil {
+		return entry{}, false
+	}
+
+	return newEntry(b, s.hexDecodingBuffer), true
+}
+
+func (s *scanner) Close() {
+	s.file.Close()
+}
+
+// entry represents one line in the /proc/net file
+type entry struct {
+	laddr []byte
+	raddr []byte
+	state []byte
+	inode []byte
+
+	// used for the purposes of decoding hex numbers
+	buffer []byte
+}
+
+func newEntry(line []byte, buffer []byte) entry {
+	e := entry{buffer: buffer}
+	iter := fieldIterator{line}
+
+	// refer to the diagram above for the line format
+	iter.skip(1)
+	e.laddr = iter.nextField()
+	e.raddr = iter.nextField()
+	e.state = iter.nextField()
+	iter.skip(5)
+	e.inode = iter.nextField()
+
+	return e
+}
+
+func (e entry) LocalAddress() (netip.Addr, uint16) {
+	return e.parseAddress(e.laddr)
+}
+
+func (e entry) RemoteAddress() (netip.Addr, uint16) {
+	return e.parseAddress(e.raddr)
+}
+
+func (e entry) Inode() int {
+	return e.atoi(e.inode)
+}
```

Comment:
Instead of saving it in raw bytes, and on demand parse it, shouldn't we save the parsed values to reduce redundant calculations and allocations everytime we call `LocalAddress` for example?